src/libopensc/card-myeid.c

```diff
@@ -131,9 +135,13 @@ static int myeid_init(struct sc_card *card)
 	u8 defatr[SC_MAX_ATR_SIZE];
 	size_t len = sizeof(defatr);
 	const char *atrp = myeid_atrs[MYEID_INFINEON_CHIP_ATR];
-
+	
 	LOG_FUNC_CALLED(card->ctx);
+
+	card->name = myeid_card_name;
```

Comment:
as far as i can see the name did not fell out of scope, it was never initialized. and to be honest, you don't need it. All information you added with this commit was already present in the card driver's name